app/src/main/java/com/github/se/icebreakrr/model/location/LocationService.kt

```diff
@@ -0,0 +1,97 @@
+package com.github.se.icebreakrr.model.location
+
+import android.annotation.SuppressLint
+import android.location.Location
+import android.os.Looper
+import android.util.Log
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationAvailability
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.Priority
+
+class LocationService(private val fusedLocationProviderClient: FusedLocationProviderClient) {
+
+  companion object {
+    // Distance in meters for triggering a location update
+    private const val UPDATE_DISTANCE_METERS = 10.0
+
+    // Location request interval in milliseconds
+    private const val REQUEST_INTERVAL_MS = 10000L
+
+    // Minimum interval between location updates in milliseconds
+    private const val MIN_UPDATE_INTERVAL_MS = 5000L
+  }
+
+  // LocationCallback instance used to handle location updates and availability changes
+  private val locationCallback: LocationCallback =
+      object : LocationCallback() {
+        override fun onLocationResult(locationResult: LocationResult) {
+          locationResult.lastLocation?.let { newLocation ->
+            // Only update if distance from last known location exceeds threshold
+            if (lastKnownLocation == null ||
+                lastKnownLocation!!.distanceTo(newLocation) > UPDATE_DISTANCE_METERS) {
+              lastKnownLocation = newLocation
+              onLocationUpdate?.invoke(newLocation) // Invoke callback with new location
+            }
+          }
+        }
+
+        override fun onLocationAvailability(availability: LocationAvailability) {
+          if (!availability.isLocationAvailable) {
+            onError?.invoke("Location service is currently unavailable.")
```

Comment:
Should add a comment to explain this function

Reply:
I believe this function is quite self-explanatory. It's designed to alert the app whenever location data becomes unavailable. Adding a comment would only make the code more cluttered.